sklearn/model_selection/tests/test_search.py

```diff
@@ -591,71 +599,190 @@ def test_param_sampler():
+                    random_search, random_search_iid)
+    all_param_keys = (param_keys_gs,) * 2 + (param_keys_rs,) * 2
+    all_n_candidates = (n_grid_points,) * 2 + (n_search_iter,) * 2
+    is_iid = [False, True] * 2
+
+    # Common tests
+    for search, param_keys, n_cand, iid in zip(all_searches, all_param_keys,
+                                               all_n_candidates, is_iid):
+        results = search.results_
+
+        # Test array type and dtype
+        assert_true(all(isinstance(results[param], np.ma.MaskedArray)
+                        for param in param_keys))
+        assert_true(all(results[key].dtype is np.dtype('O')
+                        for key in param_keys))
+        assert_false(any(isinstance(results[key], np.ma.MaskedArray)
+                         for key in score_keys))
+        assert_true(all(results[key].dtype is np.dtype('float64')
+                        for key in score_keys if not key.endswith('rank')))
+
+        # Test it contains all the required results
+        assert_array_equal(sorted(six.iterkeys(results)),
+                           sorted(param_keys + score_keys))
+        assert_true(all(results[key].shape[0] == n_cand
+                        for key in param_keys + score_keys))
+
+        # SVC is a classifier, StratifiedKFold is used
+        cv = StratifiedKFold(n_folds)
+        res_scores = np.vstack([results["test_split_%d_accuracy" % i]
+                                for i in range(n_folds)]).T
+        res_means = results["test_accuracy_mean"]
+        res_stds = results["test_accuracy_std"]
+        res_ranks = results["test_accuracy_rank"]
+        test_sample_counts = ([len(test) for _, test in cv.split(X, y)]
+                              if iid else None)
+        assert_equal(iid, search.iid)
+        assert_equal(iid, (test_sample_counts is not None))
+        means = np.average(res_scores, axis=1, weights=test_sample_counts)
+        stds = np.sqrt(np.average(
+            (res_scores - means[:, np.newaxis]) ** 2,
+            axis=1, weights=test_sample_counts))
+        assert_array_equal(res_means, means)
+        assert_array_equal(res_stds, stds)
+
+        ranks = rankdata(means, method='max')
+        # Invert the ranks. NOTE that method='max' becomes 'min' when inverted.
+        ranks = (n_cand + 1) - ranks
+        best_index = np.flatnonzero(ranks == 1)[0]
+
+        # Check the best result
+        best_score = res_means[best_index]
+
+        assert_array_equal(res_ranks, ranks)
+        assert_equal(search.best_index_, best_index)
+        assert_equal(search.best_score_, best_score)
+        assert_equal(search.best_params_,
+                     search.candidate_params_[best_index])
+
+        tied_best_params = (search.candidate_params_[i]
+                            for i in range(n_cand)
+                            if res_means[i] == best_score)
+        assert_true(search.best_params_ in tied_best_params,
```

Comment:
Surely this is true when you have already checked `assert_equal(search.best_params_, search.candidate_params_[best_index])`?